plugins/module_utils/dcim.py

```diff
@@ -142,14 +142,12 @@ def run(self):
             data["color"] = data["color"].lower()
 
         if self.endpoint == "cables":
-            cables = [
-                cable
-                for cable in nb_endpoint.all()
-                if cable.termination_a_type == data["termination_a_type"]
-                and cable.termination_a_id == data["termination_a_id"]
-                and cable.termination_b_type == data["termination_b_type"]
-                and cable.termination_b_id == data["termination_b_id"]
-            ]
+            cables = nb_endpoint.filter(
+                termination_a_type=data["termination_a_type"],
+                termination_a_id=data["termination_a_id"],
+                termination_b_type=data["termination_b_type"],
+                termination_b_id=data["termination_b_id"],
+            )
```

Comment:
Try something like this instead:

```suggestion
            cables = [
                cable
                for cable in nb_endpoint.filter(
                    termination_a_type=data["termination_a_type"],
                    termination_a_id=data["termination_a_id"],
                    termination_b_type=data["termination_b_type"],
                    termination_b_id=data["termination_b_id"],
                )
            ]
```

Reply:
Amusingly, that seems to make it worse; compare:

- No comprehension: https://github.com/nrnvgh/nautobot-ansible/actions/runs/5970422901
- With the comprehension: https://github.com/nrnvgh/nautobot-ansible/actions/runs/5977063348

I'm not convinced the comprehension is necessary since `filter()` returns a list of connections. AFAICT it was only needed in the current version so that the conditionals could be added. Side note; I ran the current code against the https://demo.nautobot.com/ instance (after having created a device and a circuit to test with, since it's the circuit case that's failing in the test harness, at least when not using a comprehension) and saw the same timeout there...and there are fewer than 1000 cables in that environment.

Reply:
I checked on my local dev environment and it does seem to work. I'm not sure why your tests are failing.

Reply:
Oh, it's specifically Nautobot 1.3 that is breaking. Let me poke around at it.

Reply:
Probably good to drop Nautobot 1.3 from our tests if we need @joewesch, and even 1.4

Reply:
Ok, so after some digging the CI failing in Nautobot v1.3 is true and this would introduce a bug if we want to continue to support v1.3. @jvanderaa are you good with us dropping v1.3 support finally?

Long story short on why it causes a bug: the filters we are using here [didn't exist before Nautobot v1.4](https://github.com/nautobot/nautobot/pull/1981). Also, if you send in a filter to v1.3 that doesn't exist it will just ignore them by default; this behavior was changed in v1.4 with the introduction of [STRICT_FILTERING](https://docs.nautobot.com/projects/core/en/stable/configuration/optional-settings/#strict_filtering). So, the combination of those filters don't exist and v1.3 just silently ignores them means the cables being returned with `nb_endpoint.filter()` will always be **all** of the cables.